piet-gpu-hal/src/vulkan.rs

```diff
@@ -1421,26 +1450,35 @@ impl Layers {
     }
 }
 
-unsafe fn choose_compute_device(
+unsafe fn choose_device(
     instance: &Instance,
     devices: &[vk::PhysicalDevice],
-    surface: Option<&VkSurface>,
+    support_graphics: bool,
 ) -> Option<(vk::PhysicalDevice, u32)> {
+    let mut desired_flags = vk::QueueFlags::COMPUTE;
+    if support_graphics {
+        desired_flags |= vk::QueueFlags::GRAPHICS;
+    }
     for pdevice in devices {
         let props = instance.get_physical_device_queue_family_properties(*pdevice);
         for (ix, info) in props.iter().enumerate() {
-            // Check for surface presentation support
-            if let Some(surface) = surface {
-                if !surface
-                    .surface_fn
-                    .get_physical_device_surface_support(*pdevice, ix as u32, surface.surface)
-                    .unwrap()
-                {
-                    continue;
-                }
-            }
+            // TODO: is this strictly necessary? We'll need a queue supporting graphics
```

Comment:
Ordinarily I am not a fan of commented out code, but see why it's done here. If I were doing it, I would replace it with a comment, something like this:

We probably want to make the support for graphics non-optional, as we will require it for render pipelines to support image scaling. In addition, when rendering to a surface, to be more technically correct we should check whether the physical device has support for that surface. This probably won't be a problem in practice, though.